webapp/components/team_general_tab.jsx

```diff
@@ -42,7 +42,7 @@ const holders = defineMessages({
     },
     openInviteTitle: {
         id: 'general_tab.openInviteTitle',
-        defaultMessage: 'Allow anyone to join this team'
+        defaultMessage: 'Allow any authenticated user to join this team'
```

Comment:
@crspeller, this is coming from a customer who thought "Allow any user to join this team" meant unauthenticated users. How about "Allow any user with an account on this server to join this team"?